src/icmp/IcmpPinger.cc

```diff
@@ -122,7 +122,7 @@ IcmpPinger::Open(void)
         return -1;
     }
 
-    x = xsend(icmp_sock, buf, strlen(buf), 0);
+    x = xsend(icmp_sock, buf, 3, 0);
```

Comment:
I agree that this PR has identified a problem with strlen() applied to a potentially non-terminated buffer. I do not know anything about Windows API in use here, not even its name, but, if for some strange reason Squid needs to echo back everything it just read from icmp_sock, then this line should become:

```suggestion
    x = xsend(icmp_sock, buf, x, 0);
```

Again, I do not claim that the above suggestion matches unnamed Windows API requirements.

P.S. Please consider adding a static_assert that validates the following hidden assumption: `PS` size does not exceed `wpi` size. This code uses `wpi`-sized `buf` to read `PS`...